File: introduction/ITP1_2_D-Circle in a Rectangle.java
import java.util.*;

public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    String[] line = sc.nextLine().split(" ");

    int w = Integer.parseInt(line[0]);
    int h = Integer.parseInt(line[1]);
    int x = Integer.parseInt(line[2]);
    int y = Integer.parseInt(line[3]);
    int r = Integer.parseInt(line[4]);

    if (r <= x && x <= w - r && r <= y && y <= h - r) {
      System.out.println("Yes");
    } else {
      System.out.println("No");
    }
  }
}

File: introduction/ITP1_2_D-Circle in a Rectangle.scala
object Main extends App {
  val sc = new java.util.Scanner(System.in)
  val line = sc.nextLine().split(" ")

  val w = line(0).toInt
  val h = line(1).toInt
  val x = line(2).toInt
  val y = line(3).toInt
  val r = line(4).toInt

  if (r <= x && x <= w - r && r <= y && y <= h - r) {
    println("Yes")
  } else {
    println("No")
  }
}
